lib/utils.js

```diff
@@ -0,0 +1,191 @@
+
+  //host:port,host2:port2, databasename
+  var hosts = hosts_path.split("/")[0];
+  var databaseName = hosts_path.split("/")[1];
+
+  //host:port host2:port2
+  var hostList = hosts.split(",");
+
+  //user, password
+  var authUser = auth.split(":")[0];
+  var authPassword = auth.split(":")[1];
+
+  result.database = {};
+  result.database.driver_options = {};
+  result.database.auth = {};
+  result.database.auth.user = authUser;
+  result.database.auth.pass = authPassword;
+  result.database.auth.source = databaseName;
+  result.database.name = databaseName;
+
+  assert.ok(authUser !== undefined);
+  assert.ok(authPassword !== undefined);
+  assert.ok(databaseName !== undefined);
+
+  //Parsing options
+  if (options) {
+    var parsedOptions = options.split(",");
+    //If options are parsed, they should have an even number of elements in the split array
+    assert.ok(parsedOptions.length > 0);
+
+    for (var i = 0; i < parsedOptions.length; i++) {
+      var splitOption = parsedOptions[i].split("=");
+      //Each option should be something=value
+      assert.ok(splitOption && splitOption.length === 2);
+      assert.ok(splitOption[0].length > 0);
+      assert.ok(splitOption[1].length > 0);
+      //Checking it does not already exist
+      assert.ok(!result.database.driver_options[splitOption[0]]);
+
+      result.database.driver_options[splitOption[0]] = splitOption[1];
+    }
+  }
+
+  var host_port;
+  if (hostList.length > 1) {
+    result.database.host = [];
+    result.database.port = [];
+    for (i = 0; i < hostList.length; i++) {
+      host_port = hostList[i].split(":");
+      assert.ok(host_port[0] && host_port[0].length > 0);
+      result.database.host.push(host_port[0]);
+      if (host_port.length > 1) {
+        result.database.port.push(host_port[1]);
+      } else {
+        result.database.port.push('27017');
+      }
+    }
+  }
+  else {
+    host_port = hostList[0].split(":");
+    assert.ok(host_port[0] && host_port[0].length > 0);
+    result.database.host = host_port[0];
+    if (host_port.length > 1) {
+      result.database.port = host_port[1];
+    } else {
+      result.database.port = '27017';
+    }
+  }
+
+
+  //Verify the config to be returned
+  assert.ok(result.database.name.length > 0);
+  assert.ok(result.database.auth.user.length > 0);
+  assert.ok(result.database.auth.pass.length > 0);
+
+
+  return result;
+};
+
+/**
+ * This will return a connection URL like:
+ * host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+ *
+ * With the intention that it can be prepended with "mongodb://[username]:[password]@"
+ * and appended with "/[database][?options]
+ *
+ * @param host This is expected in the format: "host1" or "host1,host2" or ["host1", "host2"]
+ * @param port This is expected in the format: 123 or "123" or "123,456" or ["123", "456"] or [123, 456]
+ *
+ * @returns {string} host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+ */
+exports.parseConnectionUrl = function (host, port){
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)